lib/types/orders.ts

```diff
@@ -48,7 +48,20 @@ export type OutgoingOrder = Pick<StampedOwnOrder, Exclude<keyof StampedOwnOrder,
 export type OrderIdentifier = {
   orderId: string;
   pairId: string;
-  quantity?: number;
+};
+
+export type OrderPortion = OrderIdentifier & {
+  quantity: number;
+  localId?: string;
+};
+
+export type Swap = {
```

Comment:
1. That's my goal.
2. I thought about `SwapResult` too actually, I thought `Swap` was probably clear enough but I'll change it.
3. Good question on swap failures, I wasn't thinking we would because it's not really an actionable item for the client. I'd probably lean towards leaving it as is and if we see a need for announcing swap failures in the future we can change it.